graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,268 @@
+        return field
+
+
+class _ConverterArray(_ConverterBase):  # pylint: disable=too-few-public-methods
+    def __init__(self, label_type: str, portex_type: str) -> None:
+        super().__init__(label_type, portex_type)
+        self._field_name = label_type.upper() if label_type == "rle" else f"{label_type}s"
+
+    def __call__(self, subcatalog: SubcatalogBase, labels: List[_LabelBase]) -> Dict[str, Any]:
+        items = {"type": self._portex_type}
+        items.update(_get_category_and_attribute_params(subcatalog))
+        return {"name": self._field_name, "type": "array", "items": items}
+
+
+class _ConverterArrayGeometry(_ConverterArray):  # pylint: disable=too-few-public-methods
+    def __init__(self, label_type: str, portex_type: str, label_value_level: int) -> None:
+        super().__init__(label_type, portex_type)
+        self._label_value_level = label_value_level
+
+    def __call__(self, subcatalog: SubcatalogBase, labels: List[_LabelBase]) -> Dict[str, Any]:
+        field = super().__call__(subcatalog, labels)
+
+        if len(labels) > 0:
+            value = labels
+            for _ in range(self._label_value_level):
+                value = value[0]  # type: ignore[assignment]
+            field["items"].update(_get_dtype(value))  # type: ignore[arg-type]
+
+        return field
+
+
+class _ConverterArrayKeypoints2D(_ConverterArrayGeometry):  # pylint: disable=too-few-public-methods
+    def __call__(  # type: ignore[override]
+        self, subcatalog: Keypoints2DSubcatalog, labels: List[LabeledKeypoints2D]
+    ) -> Dict[str, Any]:
+        field = super().__call__(subcatalog, labels)  # type: ignore[arg-type]
+
+        items = field["items"]
+        keypoints = subcatalog.keypoints[0]
+        items["number"] = keypoints.number
+        if hasattr(keypoints, "names"):
+            items["names"] = keypoints.names
+        if hasattr(keypoints, "skeleton"):
+            items["skeleton"] = keypoints.skeleton
+        if hasattr(keypoints, "visible"):
+            items["visible"] = keypoints.visible
+
+        return field
+
+
+_ConverterClassification("classification")
+_ConverterArrayGeometry("box2d", "label.Box2D", 2)
+_ConverterArray("box3d", "label.Box3D")
+_ConverterArrayGeometry("polygon", "label.Polygon", 3)
+_ConverterArrayGeometry("polyline2d", "label.Polyline2D", 3)
+_ConverterArrayGeometry("multi_polygon", "label.MultiPolygon", 4)
+_ConverterArrayGeometry("multi_polyline2d", "label.MultiPolyline2D", 4)
+_ConverterArrayKeypoints2D("keypoints2d", "label.Keypoints2D", 3)
+_ConverterArray("rle", "label.RLE")
+_ConverterSingle("semantic_mask", "label.file.SemanticMask")
+_ConverterInstanceMask("instance_mask", "label.file.InstanceMask")
+_ConverterPanopticMask("panoptic_mask", "label.file.PanopticMask")
+
+
+def _get_file_field(data: RemoteData, notes: Notes) -> Dict[str, Any]:
+    suffix = Path(data.path).suffix
+    for (field_name, field_type), suffixes in _FILE_TYPES_AND_SUFFIX.items():
```

Comment:
can we create a dict, which key is suffix and value is field name and value
to replace `_FILE_TYPES_AND_SUFFIX`
will it be better?